src/Compilers/Core/Portable/DiagnosticAnalyzer/AsyncQueue.cs

```diff
@@ -140,6 +146,11 @@ public void Complete()
             }
         }
 
+        public void PromiseNotToEnqueue()
```

Comment:
I would be happy to see this internal API reviewed and possible reorganized or overhauled, though we should be careful to separate out those parts that would make sense for a platform API. In any case, I want to keep the bug fix for the stabilization branch as simple and obviously correct as possible.